### DISCUSSION TITLE
Python/pytest problemMatcher doesn't seem to work?

### DISCUSSION THREAD

**Discussion by billyvg** [community/community#25393] [Actions]
<p>I’m trying to use the problemMatch defined here in <code>actions/setup-python</code>: <a href="https://github.com/actions/setup-python/blob/807b74f98ca701f414ddaa8a4187e7cffa93cbbd/.github/python.json" rel="nofollow noopener">https://github.com/actions/setup-python/blob/807b74f98ca701f414ddaa8a4187e7cffa93cbbd/.github/python.json</a></p>
<p>It doesn’t seem to work, you can see the log output here: <a href="https://github.com/getsentry/sentry/pull/19038/checks?check_run_id=711757943#step:21:130" rel="nofollow noopener">https://github.com/getsentry/sentry/pull/19038/checks?check_run_id=711757943#step:21:130</a></p>
<p>In that PR I’ve also tried with a different matcher (which doesn’t work either): <a href="https://github.com/getsentry/sentry/pull/19038/files#diff-8aa942dcc12e296fcb1cbf95181de009R7" rel="nofollow noopener">https://github.com/getsentry/sentry/pull/19038/files#diff-8aa942dcc12e296fcb1cbf95181de009R7</a></p>
<p>Testing the patterns in javascript with <code>new RegExp()</code> seem to work fine, however.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/billyvg">@billyvg</a>,<br>
Maybe you can try to reference the below file in <code>actions/setup-python</code> to add-matcher:<br>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/actions/setup-python/blob/807b74f98ca701f414ddaa8a4187e7cffa93cbbd/src/setup-python.ts" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/actions/setup-python/blob/807b74f98ca701f414ddaa8a4187e7cffa93cbbd/src/setup-python.ts" target="_blank" rel="nofollow noopener">actions/setup-python/blob/807b74f98ca701f414ddaa8a4187e7cffa93cbbd/src/setup-python.ts</a></h4>
<pre><code class="lang-ts">import * as core from '@actions/core';
import * as finder from './find-python';
import * as path from 'path';
import * as os from 'os';

async function run() {
  try {
    let version = core.getInput('python-version');
    if (version) {
      const arch: string = core.getInput('architecture') || os.arch();
      const installed = await finder.findPythonVersion(version, arch);
      core.info(`Successfully setup ${installed.impl} (${installed.version})`);
    }
    const matchersPath = path.join(__dirname, '..', '.github');
    core.info(`##[add-matcher]${path.join(matchersPath, 'python.json')}`);
  } catch (err) {
    core.setFailed(err.message);
  }
}

</code></pre>

  This file has been truncated. <a href="https://github.com/actions/setup-python/blob/807b74f98ca701f414ddaa8a4187e7cffa93cbbd/src/setup-python.ts" target="_blank" rel="nofollow noopener">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
</p>

**Comment by billyvg** [comment]
<p><a class="mention" href="/brightran">@brightran</a> Right, so by using <code>actions/setup-python</code> we should have that matcher added, correct? Now that I’m looking at the log output again, I’m not so sure if the 2nd matcher works</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/billyvg">@billyvg</a>,<br>
&gt; so by using  <strong>actions/setup-python</strong>  we should have that matcher added, correct?<br>
Yes, I think that’s it.</p>
<p>&gt; I’m not so sure if the 2nd matcher works<br>
According to the description from the docs about Problem Matchers (see <a href="https://github.com/actions/toolkit/blob/9ba7c679ad74f6a80017feeb94b7a7c19b590796/docs/problem-matchers.md#duplicate-problem-matchers" rel="nofollow noopener">here</a>),</p>
<blockquote>
<p>Registering two problem-matchers with the same owner will result in only the problem matcher registered last running.</p>
</blockquote>
<p>if you add another Problem Matchers (using the <strong>::add-matcher::</strong> command) after executing the <strong>setup-python</strong> action in your workflow, the new Problem Matchers should become the only Problem Matchers in work.</p>
